apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Logging/LoggingTest.cs

```diff
@@ -269,14 +269,11 @@ public class LoggingTest
                 var results = _polling.GetEntries(startTime, testId, 1, LogSeverity.Warning);
                 var entry = results.Single();
                 Assert.Equal(3, entry.Labels.Count);
-                var defaultLabel = entry.Labels.First();
-                Assert.Equal("some-key", defaultLabel.Key);
+                var defaultLabel = entry.Labels.Single(l => l.Key.Equals("some-key"));
```

Comment:
Alternative option:

```csharp
Assert.Equal("some-value", entry.Labels["some-key"]);
Assert.Equal("Hello, World!", entry.Labels["Foo"]);
Assert.NotEmpty(entry.Labels["Foo"]);
```

If that fails with a KeyNotFoundException, it's easy enough to see where from the stack trace.

Reply:
I like this much better, thanks for the suggestion.